mlflow/models/evaluation/base.py

```diff
@@ -151,6 +156,170 @@ def __str__(self):
         return "EvaluationMetric(" + ", ".join(parts) + ")"
 
 
+# NB: we need this function because we cannot modify the signature of
+# a class's __call__ method after the class has been defined.
+# This is also useful to distinguish between the metric signatures with different eval_fn signatures
+def _dynamically_generate_eval_metric(eval_fn, require_strict_signature=False):
+    """
+    Dynamically generate a GenAIEvaluationMetric class that can be used to evaluate the metric
+    on the given input data. The generated class is callable with a __call__ method that
+    takes the arguments specified in the signature of the eval_fn function.
+
+    Args:
+        eval_fn: the evaluation function of the EvaluationMetric.
+        require_strict_signature: (Optional) Whether the eval_fn needs to follow a strict signature.
+            If True, then the eval_fn must follow below signature:
+
+                .. code-block:: python
+
+                    def eval_fn(
+                        predictions: "pd.Series",
+                        metrics: Dict[str, MetricValue],
+                        inputs: "pd.Series",
+                        *args,
```

Comment:
```suggestion
                        **kwargs,
```

Reply:
No.. it's super weird that the original eval_fn only accepts positional args https://github.com/mlflow/mlflow/blob/2283960727fd1fdbaa09b5f5d4dcf78e8f5bb764/mlflow/metrics/genai/genai_metric.py#L504-L509

Reply:
saw the code, not even sure what `args` should be.

Reply:
`args` should be all fields defined in grading_context_columns